doc/sphinx-guides/source/container/dev-usage.rst

```diff
@@ -17,21 +17,104 @@ To test drive these local changes to the Dataverse codebase in a containerized a
 run it in addition to the necessary dependencies.
 
-Building and Running
---------------------
+| **TL;DR** *I have all that, just give me containers!*
+| Execute ``mvn -Pct clean package docker:run``, wait and continue at :ref:`dev-bootstrap`.
+
+
+.. _dev-build:
+
+Building
+--------
 
 To build the application image, run the following command, as described in :doc:`app-image`:
 
 ``mvn -Pct clean package``
 
-Now, start all the containers with a single command:
+Once this is done, you will see an image ``gdcc/dataverse:unstable`` available in your Docker cache.
+
+**Note:** This will skip any unit tests. If you have built the code before for testing, etc. you might omit the ``clean`` to
+avoid recompiling.
+
+**Note:** Also we have a ``docker-compose-dev.yml`` file, it's currently not possible to build the images without
+invoking Maven. This might change in the future.
+
+
+.. _dev-run:
+
+Running
+-------
+
+After building the app image containing your local changes to the Dataverse application, you want to run it together
+with all dependencies. There are four ways to do this (commands executed at root of project directory):
+
+.. list-table:: Cheatsheet: Running Containers
+   :widths: 15 40 45
+   :header-rows: 1
+   :stub-columns: 1
+   :align: left
+
+   * - \
+     - Using Maven
+     - Using Compose
+   * - In foreground
+     - ``mvn -Pct docker:run``
+     - ``docker compose -f docker-compose-dev.yml up``
+   * - In background
+     - ``mvn -Pct docker:start``
+     - ``docker compose -f docker-compose-dev.yml up -d``
+
+Both ways have their pros and cons:
+
+.. list-table:: Decision Helper: Fore- or Background?
+   :widths: 15 40 45
+   :header-rows: 1
+   :stub-columns: 1
+   :align: left
+
+   * - \
+     - Pros
+     - Cons
+   * - Foreground
+     - | Logs scroll by when interacting with API / UI
+       | To stop all containers simply hit ``Ctrl+C``
+     - | Lots and lots of logs scrolling by
+       | Must stop all containers to restart
+   * - Background
+     - | No logs scrolling by
+       | Easy to replace single containers
+     - | No logs scrolling by
+       | Stopping containers needs an extra command
+
+In case you want to concatenate building and running, here's a cheatsheet for you:
+
+.. list-table:: Cheatsheet: Building and Running Containers
+   :widths: 15 40 45
+   :header-rows: 1
+   :stub-columns: 1
+   :align: left
+
+   * - \
+     - Using Maven
+     - Using Compose
+   * - In foreground
+     - ``mvn -Pct package docker:run``
+     - ``mvn -Pct package && docker compose -f docker-compose-dev.yml up``
+   * - In background
+     - ``mvn -Pct package docker:start``
+     - ``mvn -Pct package && docker compose -f docker-compose-dev.yml up -d``
 
-``mvn -Pct docker:run``
+Once all containers have been started, you can check if the application was deployed correctly by checking the version
+at http://localhost:8080/api/info/version. or watch the logs.
```

Comment:
Could we adapt the compose.yaml to include a [healthcheck](https://docs.docker.com/compose/compose-file/05-services/#healthcheck) and explain users to wait until the container is `healthy`?

Reply:
Please feel free to open an issue about a healthcheck. We actually might want to not just take the api/info/version endpoint as an indication for "healthy"... I think this is beyond scope for this PR, which is docs changes only.